index.js

```diff
@@ -0,0 +1,172 @@
+const express = require('express');
+const helmet = require('helmet');
+const knex = require('knex');
+const knexConfig = require('./knexfile');
+const db = knex(knexConfig.development)
+const server = express();
+
+server.use(helmet());
+server.use(express.json());
+
+// endpoints here
+server.get('/api/cohorts', (req, res) => {
+  db.select().from('cohorts')
+    .then(cohorts => res.status(200).json(cohorts))
+    .catch(err => res.status(500).json(err))
+})
+
+server.post('/api/cohorts', (req, res) => {
+  db.select().from('cohorts')
+    .then(cohorts => res.status(200).json(cohorts))
+    .catch(err => res.status(500).json(err))
+})
+
+server.get('/api/cohorts/:id', (req, res) => {
+  const id = req.params.id;
+  db.select().where({id}).first().from('cohorts')
+    .then(cohort => {
+      if(cohort) {
+        res.status(200).json(cohort)
+      } else {
+        res.status(404).json({message: 'we cannot find that cohort!'})
+      }
+    })
+    .catch(err => res.status(500).json(err))
+})
+
+server.get('/api/cohorts/:id/students', (req, res) => {
+  const id = req.params.id;
+  db.select().where({id}).first().from('cohorts')
```

Comment:
Functions like get cohort by id rather than students by cohort id. Try the code below:
```suggestion
  db("cohorts as c")
    .join("students as s", "s.cohort_id", "=", "c.id")
    .where("c.id", id)
    .then(students => res.status(200).json(students))
    .catch(err => res.status(500).json(err));
```